url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/fri/patterns/interpreter/parsergenerator/examples/Joust.java

```diff
@@ -0,0 +1,419 @@
+		toReturn.add(ARG2);
+		toReturn.add(ARG3);
+		return toReturn;
+	}
+	
+	public Object EXPRESSION(Object name, Object leftParen, Object ARG1, Object comma, Object ARG2, Object rightParen) {
+		//return "[" + name + ", " + ARG1 + ", " + ARG2 + "]";
+		//return "[" + name + " " + ARG1 + " " + ARG2 + "]";
+		//System.out.println("EXPR ARG1: " + ARG1);
+		//System.out.println("EXPR ARG2: " + ARG2);
+		ArrayList<Object> toReturn = new ArrayList<Object>();
+		toReturn.add(name);
+		toReturn.add(ARG1);
+		String s = (String) comma;
+		if (s.equals(",") == false) {
+			toReturn.add(comma);
+		}
+		toReturn.add(ARG2);
+		return toReturn;
+	}
+	
+	public Object EXPRESSION(Object name, Object leftParen, Object ARG1, Object rightParen) {
+		//return "[" + name + ", " + ARG1 + "]";
+		//return "[" + name + " " + ARG1 + "]";
+		//System.out.println("EXPR name: " + name);
+		//System.out.println("EXPR ARG1: " + ARG1);
+		ArrayList<Object> toReturn = new ArrayList<Object>();
+		toReturn.add(name);
+		toReturn.add(ARG1);
+		return toReturn;
+	}
+	
+	public Object EXPRESSION(Object star) {
+		//return "[" + name + ", " + ARG1 + "]";
+		//return "[" + name + " " + ARG1 + "]";
+		//System.out.println("EXPR name: " + name);
+		//System.out.println("EXPR ARG1: " + ARG1);
+		ArrayList<Object> toReturn = new ArrayList<Object>();
+		toReturn.add(star);
+		return toReturn;
+	}
+	
+	public Object PARAMETERS(Object CHESSPIECE) {
+		//return "[" + CHESSPIECE + "]";
+		ArrayList<Object> toReturn = new ArrayList<Object>();
+		//System.out.println("PARAM1 CHESSPIECE: " + CHESSPIECE);
+		toReturn.add(CHESSPIECE);
+		return toReturn;
+	}
+	
+	public Object PARAMETERS(Object PARAMETERS, Object COMMA, Object CHESSPIECE) {
+		//System.out.println("PARAM2 CHESSPIECE: " + CHESSPIECE);
+		//System.out.println("PARAM2 PARAMETERS: " + PARAMETERS);
+		ArrayList<Object> p = (ArrayList<Object>) PARAMETERS;
+		p.add(CHESSPIECE);
+		return p;
+	}
+	
+	public Object VARPARAMETERS(Object VARCHESSPIECE) {
+		ArrayList<Object> toReturn = new ArrayList<Object>();
+		toReturn.add(VARCHESSPIECE);
+		System.out.println("varparams toReturn: " + toReturn);
+		return toReturn;
+	}
+	
+	public Object VARPARAMETERS(Object VARPARAMETERS, Object COMMA, Object VARCHESSPIECE) {
+		System.out.println("varparams VARPARAMETERS: " + VARPARAMETERS);
+		ArrayList<Object> p = (ArrayList<Object>) VARPARAMETERS;
+		p.add(VARCHESSPIECE);
+		return p;
+	}
+	
+	public Object CHESSPIECE(Object TYPE) {
+		return TYPE;
+	}
+	
+	public Object VARCHESSPIECE(Object TYPE, Object FACTOR) {
+		String s = (String) TYPE + (String) FACTOR;
+		System.out.println("VARCHESSPIECE: " + s);
+		return s;
+	}
+	
+	public Object COORDINATE(Object X, Object FACTOR) {
+		String s = (String) X + (String) FACTOR;
+		//System.out.println("s: " + s);
+		return s;
+	}
+	
+	public Object VARIABLE(Object AVARIABLE, Object letter) {
+		String s = (String) AVARIABLE;
+		s += (String) letter;
+		return s;
+	}
+	
+	public Object TYPE(Object piece) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.